youtube_dl/extractor/vidlii.py

```diff
@@ -0,0 +1,136 @@
+                           'fixed PS: Jan you are cool please add my video',
+            'thumbnail': 'https://www.vidlii.com/usfi/thmp/tJluaH4BJ3v.jpg',
+            'uploader': 'APPle5auc31995',
+            'url': 'https://cdn.vidlii.com/videos/tJluaH4BJ3v.mp4',
+            'uploader_url': 'https://www.vidlii.com/user/APPle5auc31995',
+            'upload_date': '20171107',
+            'categories': 'News & Politics',
+            'tags': ['Vidlii', 'Jan', 'Videogames'],
+            'duration': 212,
+            'view_count': int,
+            'comment_count': int,
+            'average_rating': float,
+            'type': 'video',
+            'ext': 'mp4'
+        }
+    }, {
+        'url': 'https://www.vidlii.com/watch?v=vBo2IcrwOkO',
+        'md5': 'b42640a596b4dc986702567d49268963',
+        'info_dict': {
+            'id': 'vBo2IcrwOkO',
+            'title': '(OLD VIDEO) i like youtube!!',
+            'description': 'Original upload date:<br />\nMarch 10th 2011<br />\nCredit goes to people who own content in the video',
+            'thumbnail': 'https://www.vidlii.com/usfi/thmp/vBo2IcrwOkO.jpg',
+            'uploader': 'MyEditedVideoSpartan',
+            'url': 'https://cdn.vidlii.com/videos/vBo2IcrwOkO.mp4',
+            'uploader_url': 'https://www.vidlii.com/user/MyEditedVideoSpartan',
+            'upload_date': '20171011',
+            'categories': 'Film & Animation',
+            'tags': None,
+            'duration': 34,
+            'view_count': int,
+            'comment_count': int,
+            'average_rating': float,
+            'type': 'video',
+            'ext': 'mp4'
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        # extract videoInfo variable for further use
+        videoInfo_dict = self._parse_json(js_to_json(self._html_search_regex(r'var videoInfo\s*=\s*({[^}]*})', webpage,
+                                                                             'videoInfo', fatal=False)), video_id)
+
+        # extract basic properties of video
+        title = (self._html_search_regex(r'<title>([^<]+?)</title>', webpage, 'title', default='') or
+                 self._html_search_meta('twitter:title', webpage, 'title', default='')).replace(" - VidLii",
+                                                                                                "") or self._html_search_regex(
+            r'<h1>(.+?)</h1>', webpage, 'title', default=None)
+
+        description = strip_or_none(get_element_by_id('des_text', webpage))
+
+        uploader = self._html_search_regex(
+            r'<div[^>]+class="wt_person"[^>]*>(?:[^<]+)<a href="/user/[^>]*?>([^<]*?)<|<img src="[^>]+?class=('
+            r'?:"avt2\s*"|\'avt2\s*\')[^>]+?alt=(?:"([^"]+?)"|\'([^\']+?)\')>', webpage, 'uploader', default=None,
+            fatal=False)
+
+        video_url = videoInfo_dict.get("src")
```

Comment:
Must be fatal.

Reply:
I do not understand what you mean. As described in class ``InfoExtractor``, the field ``uploader`` is optional, thus ``fatal = True`` is not required. 

If you refer to the ``videoInfo_dict``, the ``_html_search_regex`` for retrieving it already uses ``fatal = True``.

Please describe more precisely what you mean.

Reply:
This comment is not about uploader. You can't process without `video_url` so it's senseless to extract `src` with non-failing on missing keys `get`.

Reply:
How should the check be performed - by using ``videoInfo_dict['src']`` which automatically raises an exception if the key is not present?

Reply:
Yes.